src/order/basic.lean

```diff
@@ -160,6 +160,14 @@ def strict_mono [has_lt α] [has_lt β] (f : α → β) : Prop :=
 
 lemma strict_mono_id [has_lt α] : strict_mono (id : α → α) := λ a b, id
 
+lemma strict_mono_of_strict_mono_nat [preorder α] {f : ℕ → α} (hf : ∀n, f n < f (n + 1)) :
+  strict_mono f | n m h :=
```

Comment:
This is already there in https://leanprover-community.github.io/mathlib_docs/order/basic.html#strict_mono.nat in the same file. I think neither the old name nor the new one is great...